### PR TITLE
NEXT-28874 - Update Meteor Component Library to Storybook 7

### PR DESCRIPTION
**Pull request by NiklasLimberg**
This PR updates storybook to 7.1.1 and converts all stories to the CSF 3 format

### CODE REVIEW

File: __snapshots__/interaction-tests-feedback-indicator-sw-banner--visual-test-banner-neutral-snap.png
Comment (jleifeld) [inline __snapshots__/interaction-tests-feedback-indicator-sw-banner--visual-test-banner-neutral-snap.png:1]: Visual Tests show some differences. Here is the icon missing. It looks like that it doesn't wait for a finished layout anymore. We need a solutions for this in the new version because otherwise we can't rely on visual tests anymore

File: __snapshots__/interaction-tests-form-sw-checkbox--visual-test-inherited-snap.png
Comment (jleifeld) [inline __snapshots__/interaction-tests-form-sw-checkbox--visual-test-inherited-snap.png:1]: Here it looks like that the interaction wasn't used.

File: __snapshots__/interaction-tests-form-sw-colorpicker--visual-test-colorpicker-readonly-snap.png
Comment (jleifeld) [inline __snapshots__/interaction-tests-form-sw-colorpicker--visual-test-colorpicker-readonly-snap.png:1]: Same as the previous one. Here it looks like that the interaction wasn't used.

File: __snapshots__/interaction-tests-layout-sw-card--visual-test-extended-page-snap.png
Comment (jleifeld) [inline __snapshots__/interaction-tests-layout-sw-card--visual-test-extended-page-snap.png:1]: The font rendering changed a little bit. If you downloaded these images from the Github Actions Artifacts then it is okay 👍 If you generate them locally then I would suggest to download them and update them in the snapshots folder. Because the font rendering is different between macOS, Linux and Windows. Our truth of source is here Linux. (Same for the other screens with similar differences)

File: package.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
     "test:unit": "jest",
     "test:unit:watch": "jest --watch",
     "lint": "vue-tsc --noEmit --declaration false && eslint --ext .js,.ts,.vue src",
-    "storybook": "SHOW_INTERACTIONS=true start-storybook -p 6006",
+    "storybook": "SHOW_INTERACTIONS=true storybook dev -p 6006",
     "build-vue": "rollup -c",
-    "build-storybook": "build-storybook",
-    "test-storybook": "test-storybook --url http://127.0.0.1:6006",
+    "build-storybook": "storybook build",
+    "test-storybook": "test-storybook --url http://0.0.0.0:6006",
```
Comment (jleifeld) [inline package.json:11]: Is there a reason why you changed it back to 0.0.0.0? Sometimes the GH Runner has problems with this URL

File: src/components/form/sw-url-field/sw-url-field.stories.js [hunk 2]
```diff
@@ -19,8 +19,7 @@ const Template = (args, { argTypes }) => ({
   template: `<div>
       <sw-url-field
         v-bind="$props"
-        v-model="currentValue"
-        @change="change"
+        :value="currentValue"
```
Comment (jleifeld) [inline src/components/form/sw-url-field/sw-url-field.stories.js:22]: Why did you remove this? This would be a breaking change

File: src/components/form/sw-url-field/sw-url-field.vue [hunk 3]
```diff
@@ -55,7 +55,7 @@
         :disabled="disabled"
         @input="onInput"
         @focus="setFocusClass"
-        @blur="onBlur($event); removeFocusClass();"
+        @blur="onBlur($event);"
```
Comment (jleifeld) [inline src/components/form/sw-url-field/sw-url-field.vue:58]: Why did you remove this?

File: src/components/form/sw-url-field/sw-url-field.vue [hunk 4]
```diff
@@ -176,12 +176,12 @@ export default Vue.extend({
       this.checkInput(this.currentValue);
     },
 
-    onBlur(event: Event) {
-      // @ts-expect-error - target is defined
-      this.checkInput(event.target.value);
+    onBlur() {
```
Comment (jleifeld) [inline src/components/form/sw-url-field/sw-url-field.vue:179]: What is the reason for these changes?

File: src/components/form/sw-url-field/sw-url-field.vue [hunk 5]
```diff
@@ -176,12 +176,12 @@ export default Vue.extend({
...
+      this.checkInput(this.url);
     },
 
     checkInput(inputValue: string) {
+      console.log('checkInput', inputValue);
```
Comment (jleifeld) [inline src/components/form/sw-url-field/sw-url-field.vue:184]: Console log needs to be removed

File: src/components/form/sw-url-field/sw-url-field.vue [hunk 6]
```diff
@@ -213,6 +213,7 @@ export default Vue.extend({
     },
 
     validateCurrentValue(value: string) {
+      console.log('validateCurrentValue', value);
```
Comment (jleifeld) [inline src/components/form/sw-url-field/sw-url-field.vue:216]: Console log needs to be removed

**Review by jleifeld — CHANGES_REQUESTED** [review]

File: __snapshots__/interaction-tests-navigation-sw-tabs--visual-test-render-tabs-vertical-snap.png
Comment (NiklasLimberg) [inline __snapshots__/interaction-tests-navigation-sw-tabs--visual-test-render-tabs-vertical-snap.png:1]: This element move slightly to the right in the snapshot. It does not have any additional padding or margin compared to the previous version, so it should be fine

**Review by seggewiss — CHANGES_REQUESTED** [review]
image

File: src/components/context-menu/sw-context-button/sw-context-button.vue [hunk 7]
```diff
@@ -51,6 +52,42 @@ export default Vue.extend({
   },
 
   props: {
+    showMenuOnStartup: {
```
Comment (jleifeld) [inline src/components/context-menu/sw-context-button/sw-context-button.vue:55]: Does these new props do anything? If not, you can remove then

File: src/components/navigation/sw-tabs/sw-tabs.interactive.stories.js [hunk 8]
```diff
@@ -76,195 +77,189 @@ const tabItems = [
...
 
-export const VisualTestRenderTabs = Template.bind({});
-VisualTestRenderTabs.storyName = 'Render tabs';
-VisualTestRenderTabs.args = {
-  ...Default.args,
-  small: false,
-  defaultItem: 'item2',
-  items: tabItems.slice(0,2)
-};
+export const VisualTestRenderTabsVertical = {
```
Comment (jleifeld) [inline src/components/navigation/sw-tabs/sw-tabs.interactive.stories.js:93]: I have a guess why the snapshot could differ from the previous one. Before we used the `Default` args as the base which uses `small: true`. Now it uses `meta` which has `small: false` as default. Then the change in the snapshot would make sense

**Review by jleifeld — CHANGES_REQUESTED** [review]

File: src/components/navigation/sw-tabs/sw-tabs.interactive.stories.js (hunk 8 above)
Comment (Haberkamp) [reply src/components/navigation/sw-tabs/sw-tabs.interactive.stories.js:93]: Yes this was the issue.

**Review by seggewiss — APPROVED** [review]

**Review by jleifeld — APPROVED** [review]
